lib/aws-aurora-serverless-stack.ts

```diff
@@ -1,8 +1,83 @@
 import * as cdk from 'aws-cdk-lib';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as rds from 'aws-cdk-lib/aws-rds';
+import * as kms from 'aws-cdk-lib/aws-kms';
 import { Construct } from 'constructs';
+import { AuroraEngine, AwsAuroraServerlessStackProps } from './AwsAuroraServerlessStackProps';
+import { SecretValue } from 'aws-cdk-lib';
+import { parseVpcSubnetType } from '../utils/vpc-type-parser';
 
 export class AwsAuroraServerlessStack extends cdk.Stack {
-  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+  constructor(scope: Construct, id: string, props: AwsAuroraServerlessStackProps) {
     super(scope, id, props);
+
+    const vpc = ec2.Vpc.fromLookup(this, `${props.resourcePrefix}-VPC-Imported`, {
+      vpcId: props.vpcId,
+    });
+
+    const kmsKey = new kms.Key(this, `${props.resourcePrefix}-Aurora-KMS-Key`, {
+      enableKeyRotation: true,
+    });
+
+    const auroraPort = props.auroraEngine === AuroraEngine.AuroraPostgresql ? 5432 : 3306;
+    const auroraSecurityGroup = new ec2.SecurityGroup(this, `${props.resourcePrefix}-Aurora-Security-Group`, {
+      vpc,
+      allowAllOutbound: true,
+    });
+
+    const auroraDatabaseCluster = new rds.DatabaseCluster(this, `${props.resourcePrefix}-Aurora-Serverless`, {
+      engine: props.auroraEngine === AuroraEngine.AuroraPostgresql ?
+        rds.DatabaseClusterEngine.auroraPostgres({ version: rds.AuroraPostgresEngineVersion.VER_17_2 }) :
+        rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_3_08_0 }),
```

Comment:
**Suggestion:** Add backup retention configuration for the Aurora cluster to ensure data durability and compliance with backup requirements. The current implementation uses default settings which might not be optimal for production use.
> 🌟  **High Priority** 
 >[Category: **`Best Practice`**, Importance: **`8`**]
> **Reason:** Adding backup retention and a preferred backup window ensures data durability and compliance with backup requirements, which is crucial for production environments. This suggestion aligns with best practices for database management and improves the reliability of the system.

```suggestion
    const auroraDatabaseCluster = new rds.DatabaseCluster(this, `${props.resourcePrefix}-Aurora-Serverless`, {
      engine: props.auroraEngine === AuroraEngine.AuroraPostgresql ?
        rds.DatabaseClusterEngine.auroraPostgres({ version: rds.AuroraPostgresEngineVersion.VER_17_2 }) :
        rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_3_08_0 }),
      backup: {
        retention: cdk.Duration.days(14),
        preferredWindow: '03:00-04:00'
      },
```
- [ ]  **I have reviewed this code suggestion and addressed the relevant ones.**